### PR TITLE
[Feature][Connector-V2][ES] Support dsl filter

### PR DESCRIPTION
**Pull request by kpretty**
## Purpose of this pull request
Support DSL
More convenient data ingest, improve data real-time performance, reduce unnecessary io
```hocon
Elasticsearch {
    hosts = ["localhost:9200"]
    index = "seatunnel-*"
    source = ["_id","name","age"]
    query = {"range":{"firstPacket":{"gte":1669225429990,"lte":1669225429990}}}
}
```

## Check list

* [x] Code changed are covered with tests, or it does not need tests for reason:
* [ ] If any new Jar binary package adding in your PR, please add License Notice according
  [New License Guide](https://github.com/apache/incubator-seatunnel/blob/dev/docs/en/contribution/new-license.md)
* [x] If necessary, please update the documentation to describe the new feature. https://github.com/apache/incubator-seatunnel/tree/dev/docs
* [x] If you are contributing the connector code, please check that the following files are updated:
  1. Update change log that in connector document. For more details you can refer to [connector-v2](https://github.com/apache/incubator-seatunnel/tree/dev/docs/en/connector-v2)
  2. Update [plugin-mapping.properties](https://github.com/apache/incubator-seatunnel/blob/dev/plugin-mapping.properties) and add new connector information in it
  3. Update the pom file of [seatunnel-dist](https://github.com/apache/incubator-seatunnel/blob/dev/seatunnel-dist/pom.xml)
* [x] Update the [`release-note`](https://github.com/apache/incubator-seatunnel/blob/dev/release-note.md).

### CODE REVIEW

File: seatunnel-e2e/seatunnel-connector-v2-e2e/connector-elasticsearch-e2e/src/test/java/org/apache/seatunnel/e2e/connector/elasticsearch/ElasticsearchIT.java [hunk 1]
```diff
@@ -197,7 +197,10 @@ private List<String> readSinkData() throws InterruptedException {
                         "c_bytes",
```
Comment (hailin0) [inline seatunnel-e2e/seatunnel-connector-v2-e2e/connector-elasticsearch-e2e/src/test/java/org/apache/seatunnel/e2e/connector/elasticsearch/ElasticsearchIT.java:199]: please add e2e testcase（test job config）

Comment (kpretty) [reply seatunnel-e2e/seatunnel-connector-v2-e2e/connector-elasticsearch-e2e/src/test/java/org/apache/seatunnel/e2e/connector/elasticsearch/ElasticsearchIT.java:199]: fixed @hailin0

File: docs/en/connector-v2/source/Elasticsearch.md [hunk 2]
```diff
@@ -19,23 +19,24 @@ support version >= 2.x and < 8.x.
...
-| tls_truststore_password | string  | no       | -             |
-| common-options          |         | no       | -             |
+|          name           |  type   | required |   default value   |
+|-------------------------|---------|----------|-------------------|
+| hosts                   | array   | yes      | -                 |
+| username                | string  | no       | -                 |
+| password                | string  | no       | -                 |
+| index                   | string  | yes      | -                 |
+| source                  | array   | no       | -                 |
+| query                   | map     | no       | {"match_all": {}} |
```
Comment (hailin0) [inline docs/en/connector-v2/source/Elasticsearch.md]:
```suggestion
| query                   | json     | no       | {"match_all": {}} |
```

File: seatunnel-e2e/seatunnel-connector-v2-e2e/connector-elasticsearch-e2e/src/test/java/org/apache/seatunnel/e2e/connector/elasticsearch/ElasticsearchIT.java [hunk 3]
```diff
@@ -126,7 +127,8 @@ public void testElasticsearch(TestContainer container)
                 container.executeJob("/elasticsearch/elasticsearch_source_and_sink.conf");
         Assertions.assertEquals(0, execResult.getExitCode());
         List<String> sinData = readSinkData();
```
Comment (hailin0) [inline seatunnel-e2e/seatunnel-connector-v2-e2e/connector-elasticsearch-e2e/src/test/java/org/apache/seatunnel/e2e/connector/elasticsearch/ElasticsearchIT.java]: sinData -> sinkData

File: docs/en/connector-v2/source/Elasticsearch.md [hunk 4]
```diff
@@ -59,6 +60,11 @@ The fields of index.
 You can get the document id by specifying the field `_id`.If sink _id to other index,you need specify an alias for _id due to the Elasticsearch limit.
 If you don't config source, you must config `schema`.
 
+### query [map]
```
Comment (hailin0) [inline docs/en/connector-v2/source/Elasticsearch.md]:
```suggestion
### query [json]
```

File: docs/en/connector-v2/source/Elasticsearch.md (hunk 2 above)
Comment (kpretty) [reply docs/en/connector-v2/source/Elasticsearch.md]: @hailin0 right

File: seatunnel-e2e/seatunnel-connector-v2-e2e/connector-elasticsearch-e2e/src/test/java/org/apache/seatunnel/e2e/connector/elasticsearch/ElasticsearchIT.java (hunk 3 above)
Comment (kpretty) [reply seatunnel-e2e/seatunnel-connector-v2-e2e/connector-elasticsearch-e2e/src/test/java/org/apache/seatunnel/e2e/connector/elasticsearch/ElasticsearchIT.java]: @hailin0 fixed

File: docs/en/connector-v2/source/Elasticsearch.md (hunk 4 above)
Comment (kpretty) [reply docs/en/connector-v2/source/Elasticsearch.md]: @hailin0 you are right

**Comment by kpretty** [comment]
TBR @hailin0

File: docs/en/connector-v2/source/Elasticsearch.md [hunk 5]
```diff
@@ -59,6 +60,11 @@ The fields of index.
 You can get the document id by specifying the field `_id`.If sink _id to other index,you need specify an alias for _id due to the Elasticsearch limit.
 If you don't config source, you must config `schema`.
 
+### query [json]
+
+Elasticsearch DSL.
+You can control the range of data read
```
Comment (TaoZex) [inline docs/en/connector-v2/source/Elasticsearch.md]:
```suggestion
You can control the range of data read.
```

**Review by TaoZex — COMMENTED** [review]
LGTM

**Review by hailin0 — APPROVED** [review]

**Review by EricJoy2048 — APPROVED** [review]

**Pull request merged**